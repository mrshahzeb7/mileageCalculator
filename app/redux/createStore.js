import {applyMiddleware, compose, createStore} from 'redux';

export default (rootReducer, rootSaga) => {
  const middleware = [];
  const enhancers = [];

  enhancers.push(applyMiddleware(...middleware));

  const createAppropriateStore = createStore;

  const store = createAppropriateStore(rootReducer, compose(...enhancers));

  return {
    store,
  };
};
