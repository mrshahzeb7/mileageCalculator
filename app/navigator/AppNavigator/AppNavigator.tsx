import React from 'react';
import {StyleSheet, View} from 'react-native';
import {useAppSelector} from '../../redux/hooks';
import {Home, TimeLine} from '../../screens';
import {routeNames} from '../routeNames';

interface Props {}

const screenComponents = {
  [routeNames.home]: Home, // Use the imported Home component
  [routeNames.timeLine]: TimeLine, // Use the imported TimeLine component
};

export const AppNavigator: React.FC<Props> = () => {
  const selectedScreen = useAppSelector(
    state => state.navigation.selectedTabScreen,
  );

  const SelectedComponent = screenComponents[selectedScreen];

  return (
    <View style={styles.container}>
      {SelectedComponent && <SelectedComponent />}
    </View>
  );
};

// styles
const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
