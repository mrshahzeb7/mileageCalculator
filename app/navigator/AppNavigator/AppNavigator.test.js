import 'react-native';
import React from 'react';
import {AppNavigator} from './AppNavigator';

// Note: import explicitly to use the types shiped with jest.
import {it} from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import createStore from '../../redux';

import {Provider} from 'react-redux';
import {
  SafeAreaProvider,
  initialWindowMetrics,
} from 'react-native-safe-area-context';
const store = createStore();

const Parent = () => {
  return (
    <Provider store={store}>
      <SafeAreaProvider initialMetrics={initialWindowMetrics}>
        <AppNavigator />
      </SafeAreaProvider>
    </Provider>
  );
};

it('AppNavigator renders correctly', () => {
  const tree = renderer.create(<Parent />).toJSON();
  expect(tree).toMatchSnapshot();
});
