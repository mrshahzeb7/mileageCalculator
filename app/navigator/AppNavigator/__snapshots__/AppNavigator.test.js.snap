// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNavigator renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
/>
`;
